x/emissions/module/rewards/rewards.go

```diff
@@ -259,7 +259,42 @@ func EmitRewards(ctx sdk.Context, k keeper.Keeper, block BlockHeight) error {
 		if err != nil {
 			return errors.Wrapf(err, "failed to delete topic reward nonce")
 		}
+
+		// Get oldest unfulfilled nonce - delete everything behind it
```

Comment:
Consider abstracting much of this new code into a new function to make it easier to test.